### DISCUSSION TITLE
How to install umi cli ?

### DISCUSSION THREAD

**Discussion by hyzx86** [umijs/umi#10756] [Q&A] [answered]
https://umijs.org/docs/tutorials/getting-started

官方文档中似乎没有提及任何安装 umicli 的示例？？
上来就是 pnpm umi g ？？

**Comment by xiaohuoni** [comment, marked as answer]
你要先装 umi 吧，装了 umi 就都有了。
前面的步骤不是让你新建个 umi 项目，不就有 umi g 了
https://umijs.org/docs/tutorials/getting-started#%E5%88%9B%E5%BB%BA%E9%A1%B9%E7%9B%AE
```
 pnpm dlx create-umi@latest
```

**Reply by hyzx86** [reply]
node  v16.18.0


![image](https://user-images.githubusercontent.com/15613121/225252992-82b354ee-7a8a-40f7-9e27-487f2eb35e4b.png)


但是这个命令似乎是创建项目的？ yarn 好像能跑 
我在 antdpro （github 最新） 项目上直接用 umi xxx 的命令 也不能用

**Reply by hyzx86** [reply]
对啊 。。怎么装 umi 呢~ 
我这电脑上貌似有个 umi3.5 的版本~~  装不了4了

**Reply by hyzx86** [reply]
你的意思是, 直接在项目目录里 执行 umi 命令 那就是 当前项目的 umi 对吗？
这个是最新版 antdpro ，它里面是有依赖umi的，但是我敲umi -v出来 是3.5 ~~

![image](https://user-images.githubusercontent.com/15613121/225260390-bf14b0c9-e5e2-4752-826c-f3f597a60fd2.png)

**Reply by xiaohuoni** [reply]
> 我在 antdpro （github 最新） 项目上直接用 umi xxx 的命令 也不能用

pro 项目 用的是 max 命令，你试试 npx max xxx

**Reply by xiaohuoni** [reply]
你直接用 umi xxx 用的是全局上的 umi，不是你当前项目的。

**Reply by hyzx86** [reply]
是啊 。。但是  umi  这个cli 怎么手动全局安装呢？ 
pnpm umi -g 
or
pnpm umijs -g 


好吧 。。我试出来了 是 
```
npm i umi -g 
```

建议加到官方文档~~~
